File: Primes.java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

public class Primes {
    public static void main(String[] args) {
        int n = 1000000000;
        boolean[] primes = new boolean[n];
        for (int i = 0; i < n; i++) {
            primes[i] = true;
        }
        primes[0] = false;
        primes[1] = false;

        int limit = (int) Math.sqrt(n);

        for (int i = 2; i < limit; i++) {
            if (primes[i]) {
                for (int j = i * i; j < n; j += i) {
                    primes[j] = false;
                }
            }
        }

        try (BufferedWriter writer = new BufferedWriter(new FileWriter("primes.txt"))) {
            for (int i = 2; i < n; i++) {
                if (primes[i]) {
                    System.out.println(i);
                    writer.write(i + "\n");
                }
            }
        } catch (IOException e) {
            System.err.println("Error writing to file: " + e.getMessage());
        }
    }
}

File: primes.cpp
#include <iostream>
#include <cmath>
#include <fstream> // For file handling

using namespace std;

int main()
{
    int n = 1000000000;
    bool primes[n];
    for (int i = 0; i < n; i++)
    {
        primes[i] = true;
    }
    primes[0] = false;
    primes[1] = false;

    int limit = sqrt(n);

    for (int i = 2; i < limit; i++)
    {
        if (primes[i]) // Only proceed if 'i' is still prime
        {
            for (int j = i * i; j < n; j += i)
            {
                primes[j] = false;
            }
        }
    }

    ofstream outFile("primes.txt"); // Open a file for writing
    if (!outFile)
    {
        cerr << "Error opening file!" << endl;
        return 1;
    }

    for (int i = 2; i < n; i++)
    {
        if (primes[i])
        {
            cout << i << endl;
            outFile << i << endl; // Write the prime number to the file
        }
    }

    outFile.close(); // Close the file
    return 0;
}
